subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossVersionPerformanceTestRunner.groovy

```diff
@@ -138,92 +138,90 @@ class AbstractCrossVersionPerformanceTestRunner extends PerformanceTestSpec {
-            if (version == currentBaseVersion) {
-                // current version is run by default, skip adding it to baseline
-                continue
-            }
-            if (version == 'last') {
-                addMostRecentRelease = false
-                resultBaselineVersions.add(mostRecentRelease)
-                continue
-            }
-            if (version == 'nightly') {
-                addMostRecentRelease = false
-                resultBaselineVersions.add(LatestNightlyBuildDeterminer.latestNightlyVersion)
-                continue
-            }
-            if (version == 'none') {
-                return Collections.emptyList()
-            }
-            if (version == 'defaults') {
-                throw new IllegalArgumentException("'defaults' shouldn't be used in target versions.")
-            }
-            def releasedVersion = findRelease(releases, version)
-            def versionObject = GradleVersion.version(version)
-            if (minimumBaseVersion != null && baseVersionLowerThanMinimunRequirement(versionObject, minimumBaseVersion)) {
-                //this version is not supported by this scenario, as it uses features not yet available in this version of Gradle
-                continue
-            }
-            if (releasedVersion) {
-                resultBaselineVersions.add(releasedVersion.version.version)
-            } else if (versionObject.snapshot || isRcVersion(versionObject)) {
-                // for snapshots, we don't have a cheap way to check if it really exists, so we'll just
-                // blindly add it to the list and trust the test author
-                // Only active rc versions are listed in all-released-versions.properties that ReleasedVersionDistributions uses
-                addMostRecentRelease = false
-                resultBaselineVersions.add(version)
-            } else {
-                throw new RuntimeException("Cannot find Gradle release that matches version '$version'")
-            }
+        if (resolvedVersions.isEmpty() || addMostRecentRelease(overrideBaselinesProperty, versions)) {
+            // Always include the most recent final release if we're not testing against a nightly or a snapshot
+            resolvedVersions.add(releases.mostRecentRelease.version.version)
         }
 
-        if (allBaselineVersionsAreFiltered(resultBaselineVersions, versions, minimumBaseVersion)) {
-            throw new RuntimeException("All versions are filtered out by minimumBaseVersion!")
-        }
+        resolvedVersions.removeAll { !versionMeetsLowerBaseVersionRequirement(it, minimumBaseVersion) }
 
-        if (resultBaselineVersions.empty || addMostRecentRelease) {
-            // Always include the most recent final release if we're not testing against a nightly or a snapshot
-            resultBaselineVersions.add(mostRecentRelease)
+        if (resolvedVersions.isEmpty()) {
+            Assume.assumeFalse(isHistoricalPerformanceTest(versions))
```

Comment:
Please add a message why the test has been skipped here.

Reply:
Done.